src/util/system.cpp

```diff
@@ -1132,6 +1132,25 @@ int GetNumCores()
     return std::thread::hardware_concurrency();
 }
 
+void TraceThread(const char* thread_name, std::function<void()> thread_func)
```

Comment:
As you move this anyway, what about moving threadnames.h to thread.h and adding this to that file? Advantage would be that stuff like `src/index/base.cpp` that doesn't need the whole system.h can just include the slim thread.h

Reply:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/19064#issuecomment-635240245).